### PR TITLE
UCP: modifications in ucp_listener_create (CM flow).

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

File: src/ucp/core/ucp_listener.c [hunk 1]
```diff
@@ -306,14 +311,14 @@ ucp_listen_on_cm(ucp_listener_h listener, const ucp_listener_params_t *params)
         if (status != UCS_OK) {
             goto err_destroy_listeners;
         }
-    }
 
-    /* return the status of the last call of uct_listener_create if no listener
-       was created */
-    return (listener->num_rscs > 0) ? UCS_OK : status;
+        return UCS_OK;
```
Comment (yosefe) [inline src/ucp/core/ucp_listener.c:315]: on line 308:
```
if (listener->num_rcs == 0) {
   ucs_assert(status != UCS_OK);
   goto err_destroy_listeners;
}

stats =ucs_sockaddr_copy(...)
if (status != UCS_OK) {
  ...
}

return UCS_OK;
```
to unify error flow

**Comment by alinask** [comment]
@yosefe Can you take a look please?

**Comment by alinask** [comment]
bot:pipe:retest

**Comment by alinask** [comment]
bot:pipe:retest

**Comment by alinask** [comment]
@yosefe is this good to go?

**Pull request merged**